app/models/lookup_keys/puppetclass_lookup_key.rb

```diff
@@ -41,6 +41,10 @@ def puppet?
   def check_override_selected
     return if (changed - ['description', 'override']).empty?
     return if override?
+    return if !override? && override_changed? && (changed - ['merge_overrides',
```

Comment:
Yea, my concern was systems which already have override = false and merge_overrides = true. They're going to hit the validation error if they try to update their puppet classes. I suppose I could write a migration though?